src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/NestedFieldQueryIT.java

```diff
@@ -471,19 +436,28 @@ protected U featureValueOf(T actual) {
                 Query Utility to Fetch Response for SQL
      ***********************************************************/
 
-    private SearchResponse query(String select, String... statements) {
+    private SearchResponse query(String select, String... statements) throws IOException {
         return execute(select + " " + FROM + " " + String.join(" ", statements));
     }
 
-    private SearchResponse execute(String sql) {
-        SearchDao searchDao = MainTestSuite.getSearchDao();
-        try {
-            SqlElasticRequestBuilder result = searchDao.explain(sql).explain();
-            return (SearchResponse) result.get();
-        }
-        catch (SqlParseException | SQLFeatureNotSupportedException e) {
-            throw new IllegalStateException("Query failed: " + sql, e);
-        }
+    private SearchResponse execute(String sql) throws IOException {
```

Comment:
np: this function is now in too many places, would be good to re-use. You can add a TODO for now at least to have some way of tracking.

Reply:
Sure, will add a comment for TODO.